sts-mock/src/token/tokenEncrypter/tests/tokenEncrypter.test.ts

```diff
@@ -0,0 +1,220 @@
+
+      const result = await tokenEncrypter.encrypt(dummyJwt);
+
+      expect(result.isError).toBe(true);
+      expect(result.value).toStrictEqual({
+        errorMessage: "Not a valid JWKS",
+        errorCategory: "SERVER_ERROR",
+      });
+    });
+  });
+
+  describe("Given 'keys' is not an array", () => {
+    it("Returns an error response", async () => {
+      jest.spyOn(global, "fetch").mockImplementation(() =>
+        Promise.resolve({
+          status: 200,
+          ok: true,
+          text: () => Promise.resolve('{"keys":"notAnArray"}'),
+        } as Response),
+      );
+
+      const result = await tokenEncrypter.encrypt(dummyJwt);
+
+      expect(result.isError).toBe(true);
+      expect(result.value).toStrictEqual({
+        errorMessage: "Not a valid JWKS",
+        errorCategory: "SERVER_ERROR",
+      });
+    });
+  });
+
+  describe("Given the JWKS does not contain an encryption JWK", () => {
+    it("Returns an error response", async () => {
+      jest.spyOn(global, "fetch").mockImplementation(() =>
+        Promise.resolve({
+          status: 200,
+          ok: true,
+          text: () => Promise.resolve('{"keys":[{"kty":"RSA","use":"sig"}]}'),
+        } as Response),
+      );
+
+      const result = await tokenEncrypter.encrypt(dummyJwt);
+
+      expect(result.isError).toBe(true);
+      expect(result.value).toStrictEqual({
+        errorMessage: "No encryption key in JWKS",
+        errorCategory: "SERVER_ERROR",
+      });
+    });
+  });
+
+  describe("Given the encryption JWK is invalid and therefore cannot be parsed as a KeyObject", () => {
+    it("Returns an error response", async () => {
+      jest.spyOn(global, "fetch").mockImplementation(() =>
+        Promise.resolve({
+          status: 200,
+          ok: true,
+          text: () => Promise.resolve('{"keys":[{"kty":"RSA","use":"enc"}]}'),
+        } as Response),
+      );
+
+      const result = await tokenEncrypter.encrypt(dummyJwt);
+
+      expect(result.isError).toBe(true);
+      expect(result.value).toStrictEqual({
+        errorMessage: "Error creating public encryption key",
+        errorCategory: "SERVER_ERROR",
+      });
+    });
+  });
+
+  describe("Given the encryption JWK is invalid as it's missing required fields", () => {
+    it("Returns an error response", async () => {
+      jest.spyOn(global, "fetch").mockImplementation(() =>
+        Promise.resolve({
+          status: 200,
+          ok: true,
+          text: () => Promise.resolve('{"keys":[{"kty":"RSA","use":"enc"}]}'),
+        } as Response),
+      );
+
+      const result = await tokenEncrypter.encrypt(dummyJwt);
+
+      expect(result.isError).toBe(true);
+      expect(result.value).toStrictEqual({
+        errorMessage: "Error creating public encryption key",
+        errorCategory: "SERVER_ERROR",
+      });
+    });
+  });
+
+  describe("Given an error happens trying to encrypt the JWT (e.g. JWK type is not compatible with encryption algorithm)", () => {
```

Comment:
It says "e.g. JWK type is not compatible with encryption algorithm"

Is this the specific reason THIS test is failing? If it is, are there other scenarios we should test specifically?

If not would it be better worded as Given there is an unexpected error trying to encrypt etc.. 

The wording of eg is just throwing me a little

Reply:
I hear you. But we don't need to test other behaviours because we would be testing the library (`jose`) and hopefully their developers have done that. 😅 Ideally I would mock `jose` to throw an error (so that the error really looks like something unexpected has happened) but because we don't particularly like using `jest.mocks` I thought I'd just cause the error myself, and I did that by making the key type not compatible with the encryption algorithm because I knew the library would "complain". I thought I'd include this in the test title to make it clear, but perhaps a comment is better?

This error would not happen in reality unless we modify the code.

Reply:
```suggestion
// This tests what happens in our codebase if jose throws an error. It does not test every scenario for which jose might throw an error as this would be testing the underlying functionality of jose. The example below is a scenario where the JWK type is not compatible with the encryption algorithm, which is one scenario for which jose might throw an error.

describe("Given there is an unexpected error encrypting the JWT", () => {
```

@jmooney-dd - I think this is what we're trying to convey here - very keen we don't start testing jose. Does this help?

Reply:
Yeah this all makes sense, thanks both for explaining!

Reply:
Test description updated and a comment added.